File: CS-Examples/01_Quick guide/CreateFiftyExcelFiles.cs
using Spire.Xls;
using System;
using System.Windows.Forms;

namespace CreateFiftyExcelFiles
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            // Get the current date and time
            DateTime start = DateTime.Now;

            // Create 50 workbooks with 5 sheets each
            for (int n = 0; n < 50; n++)
            {
                // Create a new workbook object
                Workbook workbook = new Workbook();

                // Create 5 empty sheets in the workbook
                workbook.CreateEmptySheets(5);

                // Iterate through each sheet in the workbook
                for (int i = 0; i < 5; i++)
                {
                    // Get the reference to the current sheet
                    Worksheet sheet = workbook.Worksheets[i];

                    // Set the name of the sheet based on the index
                    sheet.Name = "Sheet" + i.ToString();

                    // Populate the sheet with data
                    for (int row = 1; row <= 150; row++)
                    {
                        for (int col = 1; col <= 50; col++)
                        {
                            // Set the cell value to a combination of the current row and column numbers
                            sheet.Range[row, col].Text = "row" + row.ToString() + " col" + col.ToString();
                        }
                    }
                }

                // Specify the filename for the resulting Excel file, using the iteration number
                workbook.SaveToFile("Workbook" + n + ".xlsx", ExcelVersion.Version2010);

                // Dispose of the workbook object
                workbook.Dispose();
            }

            // Get the current date and time after the operation
            DateTime end = DateTime.Now;

            // Calculate the time taken by subtracting the start time from the end time
            TimeSpan time = end - start;

            // Show a message box with the success message and the time consumed
            MessageBox.Show("50 File(s) have been created successfully! \n" + "Time consumed (Seconds): " + time.TotalSeconds.ToString());
          
        }


        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
    }
}

File: VB-Examples/01_Quick guide/CreateFiftyExcelFiles.vb
Imports Spire.Xls
Imports System.ComponentModel
Imports System.Text

Namespace CreateFiftyExcelFiles
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Record the start time.
            Dim start As Date = Date.Now

            For n As Integer = 0 To 49
                ' Create a new Excel workbook object.
                Dim workbook As New Workbook()
                workbook.CreateEmptySheets(5)
                For i As Integer = 0 To 4
                    ' Get the worksheet at the current index.
                    Dim sheet As Worksheet = workbook.Worksheets(i)
                    ' Set the worksheet name.
                    sheet.Name = "Sheet" & i.ToString()

                    ' Fill in data by iterating through rows and columns.
                    For row As Integer = 1 To 150
                        For col As Integer = 1 To 50
                            ' Set text content in the cell.
                            sheet.Range(row, col).Text = "row" & row.ToString() & " col" & col.ToString()
                        Next col
                    Next row
                Next i
                ' Save the workbook to a file with a unique name.
                workbook.SaveToFile("Workbook" & n & ".xlsx", ExcelVersion.Version2010)

                ' Release the resources used by the workbook
                workbook.Dispose()
            Next n
            ' Record the end time.
            Dim [end] As Date = Date.Now
            ' Calculate the time difference.
            Dim time As TimeSpan = [end].Subtract(start)
            MessageBox.Show("50 File(s) have been created successfully! " & vbLf & "Time consumed (Seconds): " & time.TotalSeconds.ToString())

        End Sub

		Private Sub FileViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace
